File: blue-serial/src/main/java/io/github/chayanforyou/blueserial/BluetoothConnection.java
package io.github.chayanforyou.blueserial;

import java.io.IOException;
import java.util.UUID;

public interface BluetoothConnection {
    boolean isConnected();

    /// Connects to given device by hardware address
    void connect(String address, UUID uuid) throws IOException;

    /// Connects to given device by hardware address (default UUID used)
    void connect(String address) throws IOException;

    /// Disconnects current session (ignore if not connected)
    void disconnect();

    /// Writes to connected remote device
    void write(byte[] data) throws IOException;
}

File: blue-serial/src/main/java/io/github/chayanforyou/blueserial/BluetoothConnection.kt
package io.github.chayanforyou.blueserial

import java.io.IOException
import java.util.UUID

interface BluetoothConnection {
    fun isConnected(): Boolean

    /// Connects to given device by hardware address
    @Throws(IOException::class)
    suspend fun connect(address: String, uuid: UUID)

    /// Connects to given device by hardware address (default UUID used)
    @Throws(IOException::class)
    suspend fun connect(address: String)

    /// Disconnects current session (ignore if not connected)
    fun disconnect()

    /// Writes to connected remote device
    @Throws(IOException::class)
    fun write(data: ByteArray)
}
